kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/ProtocolCounter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.testplugins;
+
+import java.util.EnumMap;
+import java.util.Set;
+
+import org.apache.kafka.common.protocol.ApiKeys;
+
+import io.kroxylicious.proxy.filter.Filter;
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.filter.FilterFactoryContext;
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+import io.kroxylicious.proxy.plugin.Plugins;
+import io.kroxylicious.proxy.testplugins.ProtocolCounter.Config;
+
+@Plugin(configType = Config.class)
+public class ProtocolCounter implements FilterFactory<Config, Config> {
```

Comment:
javadoc describing that it counts configured RPCs and emits count headers on produce requests would be handy